src/test/java/com/palantir/gradle/versions/ConflictSafeLockFileTest.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.versions;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.palantir.gradle.versions.lockstate.LockState;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.regex.Pattern;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+class ConflictSafeLockFileTest {
+
+    private static final Path CURRENT_SAMPLE_LOCK_FILE = Paths.get("src/test/resources/sample-versions-current.lock");
```

Comment:
perhaps it doesn't matter as you might be able to infer it from these tests, but I think you're missing a:
* read `pre-blank-lines.lock`
* write `blank-lines.lock`
* => it's what you want

I imagine this is what the "migration" we care about should do. Your tests either check you can read both formats, and can write and read new formats. Think it would be relatively harmless to add another assertion to `should_parse_a_lock_file_pre_blanks_lines_successfully` that just verifies the spacing

Reply:
I've updated the tests to this